Aleksey-Timoschenko/nodejs-aws-shop-be
# Task 7

**7.1:**
1. Create a new service called authorization-service at the same level as Product and Import services with its own AWS CDK Stack - **DONE**
2. Create a lambda function called basicAuthorizer under the Authorization Service - **DONE**
3. This lambda should have at least one environment variable with the following credentials: {yours_github_account_login}=TEST_PASSWORD - **DONE**
4. This basicAuthorizer lambda should take Basic Authorization token, decode it and check that credentials provided by token exist in the lambda environment variable - **DONE**
5. This lambda should return 403 HTTP status if access is denied for this user (invalid authorization_token) and 401 HTTP status if Authorization header is not provided - **DONE**

**7.2:**
1. Add Lambda authorization to the /import path of the Import Service API Gateway - **DONE**
2. Use your basicAuthorizer lambda as the Lambda authorizer - **DONE**

**7.3:**
1. Request from the client application to the /import path of the Import Service should have Basic Authorization header:
Authorization: Basic {authorization_token} - **DONE**
2. Client should get authorization_token value from browser localStorage - **DONE**

**Additional:**
1. Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file. - **DONE**

**Links**
- FE PR - https://github.com/Aleksey-Timoschenko/nodejs-aws-shop-react/pull/6
- App link - https://d1akx69qkmi8fp.cloudfront.net/ 

**Testing:**
- 401 Unauthorized
1. Use application link - https://d1akx69qkmi8fp.cloudfront.net/ 
2. Open 'manage products' page 
3. Click 'choose file' (be sure that you choose valid csv file)
4. Before clicking on 'upload file' open 'applications tab' in dev tools and remove 'authorization_token' field from local storage (or skip this step if there is no such field)
5. Click 'upload file'
6. Authorization error: 401 Unauthorized should appear

- 403 Forbidden
1. Use application link - https://d1akx69qkmi8fp.cloudfront.net/ 
2. Open 'manage products' page 
3. Click 'choose file' (be sure that you choose valid csv file)
4. Before clicking on 'upload file' open 'applications tab' in dev tools and set 'authorization_token' field to local storage with any random value (eg 123)
5. Click 'upload file'
6. Authorization error: 403 Forbidden should appear

- 200 OK
1. Use application link - https://d1akx69qkmi8fp.cloudfront.net/ 
2. Open 'manage products' page 
3. Click 'choose file' (be sure that you choose valid csv file)
4. Before clicking on 'upload file' open 'applications tab' in dev tools and set 'authorization_token' field to local storage with value YWxla3NleXRpbW9zY2hlbmtvPVRFU1RfUEFTU1dPUkQ= (it's an equivalent for base64 encoded string alekseytimoschenko=TEST_PASSWORD)
5. Click 'upload file'
6. Request should passed without errors

Video Record:
Link - https://www.veed.io/view/42c59f39-db38-4638-81cc-fa9d22786ede?sharingWidget=true&panel=share
